docs/guides/admin/docs/workflowoperationhandlers/select-streams-woh.md

```diff
@@ -0,0 +1,80 @@
+
+The workflow operation will use workflow properties set by the Opencast video editor to determine which tracks should be
+selected for further processing and add them to the media package based on `target-flavor` and `target-tags`.
+
+
+Parameter Table
+---------------
+
+Configuration Key | Example   | Description
+:-----------------|:----------|:-----------
+source-flavor\*   | */source  | The flavor of the track(s) to use as a source input
+target-flavor\*   | */work    | The flavor of the target track(s)
+target-tags       | download  | The tags applied to all target tracks
+audio-muxing      | force     | Move single-audio media packages to a specific track (see below)
+force-target      | presenter     | Target track for the `force` setting for `audio-muxing`
+
+\* mandatory configuration key
+
+Audio Muxing
+-----------------
+
+The optional `audio-muxing` parameter has three possible values: `none` (same as omitting the option), `force` and
+`duplicate`.
+
+### `none` ###
+
+If `none` is specified or the option is omitted, tracks are taken from the specified `source-flavor` and are edited
+according to the selections in video editor’s “Tracks” panel. The resulting tracks are stored in the corresponding `
+target-flavor` and `target-tags` are applied.
+
+Editing in the “Tracks” panel usually means removing video or audio streams from tracks. However, if there are multiple
+video and audio streams, but only one video stream and one audio stream is “non-hidden”, then these two streams will be
+muxed together into a single track.
+
+### `force` ###
+
+The parameter value `force` only applies to media packages that have exactly one non-hidden audio stream. For media
+packages without an audio stream or with more than one audio stream, the behavior is the same as if the parameter were
+omitted. The same applies to media packages for which there is only one audio stream, and it already belongs to the
+track with flavor type given by `force-target` (or `presenter` if that parameter is omitted).
+
+If, however, there is only one non-hidden audio stream and it does *not* belong to the track given by `force-target`,
+then the WOH will “move” the audio stream to this target track. Specifically, it will mux the video stream of
+`force-target` with the audio stream it found. Then, it removes the audio stream from the original track.
+
+For example, if there is a media package with two tracks, “presenter” and “presentation”, and the audio stream of
+“presenter” is hidden. Then the WOH will mux presenter’s video stream and presentations audio stream and store it in
+presenter’s place. It will also remove the audio stream from “presentation”.
+
+### `duplicate` ###
+
+The parameter value `duplicate` only applies to media packages that have exactly one non-hidden audio stream and no
+hidden video streams. For media packages without an audio stream or with more than one audio stream, the behavior
```

Comment:
Why does this only work for MPs with non-hidden video streams?  If it's just duplicating from source to N outputs, why can't it handle N-1 outputs?

Reply:
Honest reason: Because the code is really complicated as it is. ;) And it's not really properly prepared for videos with more than two tracks (yeah, "tracks" is correct this time :D), which I have stated in the long commentary before the if/else barrage. The plan was to improve on this WOH if somebody has N>2 as a requirement.
If you think it's wrong to merge it in this state, I'll of course rewrite it.

Reply:
@pmiddend & @gregorydlogan Please note that we do explicitly not have the requirement that this code can deal with more than two tracks and it was never our plan to add support for N tracks. The feature is, however, implemented in a way that would allow other Adopters to continue to work on this in case they would want to implement support for N tracks.